sui_core/src/authority/authority_store.rs

```diff
@@ -17,20 +18,56 @@ use typed_store::rocks::{DBBatch, DBMap, TypedStoreError};
 
 use typed_store::{reopen, traits::Map};
 
-pub type AuthorityStore = SuiDataStore<false>;
+pub type AuthorityStore = SuiDataStore<true>;
 #[allow(dead_code)]
-pub type ReplicaStore = SuiDataStore<true>;
+pub type ReplicaStore = SuiDataStore<false>;
 
 const NUM_SHARDS: usize = 4096;
 
 /// The key where the latest consensus index is stored in the database.
 // TODO: Make a single table (e.g., called `variables`) storing all our lonely variables in one place.
 const LAST_CONSENSUS_INDEX_ADDR: u64 = 0;
 
+#[derive(Serialize, Deserialize)]
+pub enum RawOrSignedData<D, const SIGNED: bool> {
```

Comment:
I'm not sure I get why the SIGNED bool type exists.    The point of an enum is that it could be either signed or raw, and it is decided at runtime.   Right now we have two possible `RawOrSignedData` types for a given D: either SIGNED=true, or SIGNED=false, and in either case you can have two different enum values.  This is redundant!